### PR TITLE
Updates ESLint to `2.13.1` from `2.7.0`

### PR DESCRIPTION
**Pull request by anselmbradford**
## Changes

- Updates ESLint to `2.13.1` from `2.7.0`
- Adds `init-declarations`, `max-lines`, `object-curly-newline`, `newline-before-return`, and `newline-per-chained-call` rules, but does not enforce them.
- Enforces `no-prototype-builtins`, `no-unsafe-finally`, `no-duplicate-imports`, `unicode-bom`, `no-useless-computed-key`, `no-useless-rename`, `rest-spread-spacing`, and `no-restricted-globals` rules.
- Warns on `no-useless-escape`, `keyword-spacing`, `max-statements-per-line`, `no-mixed-operators`, `object-property-newline`
- Adds options to `no-confusing-arrow`, `no-magic-numbers`, and `no-unused-vars`. Only `no-confusing-arrow` option has an effect and relaxes the rule.
- Minor code cleanup to fix linter warnings.

## Testing

- `./setup.sh`
- `gulp lint` should have no errors.

## Review

- @jimmynotjim 
- @sebworks 
- @KimberlyMunoz

### CODE REVIEW

File: cfgov/unprocessed/js/modules/form-validation.js [hunk 1]
```diff
@@ -33,7 +33,8 @@ function _sendError( elem, field ) {
   var type;
 
   for ( var key in field.status ) {
-    if ( field.status.hasOwnProperty( key ) && field.status[key] === false ) {
+    if ( {}.hasOwnProperty.call( field.status, key ) &&
```
Comment (sebworks) [inline cfgov/unprocessed/js/modules/form-validation.js]: is it possible to just make `hasOwnProperty` a utility?

**Comment by sebworks** [comment]
💯, but would love not to have to repeat `{}.hasOwnProperty.call`.

**Comment by anselmbradford** [comment]
@sebworks preference on relaxing eslint.org/docs/rules/no-prototype-builtins vs making it a utility?

**Comment by sebworks** [comment]
I would say just disable the rule as I think it's an edge case.

**Comment by anselmbradford** [comment]
@sebworks cool, updated. Please check again! Thanks!

**Comment by sebworks** [comment]
👍

**Pull request merged**